paasta_tools/contrib/service_shard_update.py

```diff
@@ -214,6 +274,56 @@ def main(args):
                         instance_config["cpus"] = args.cpus
                     if args.mem is not None:
                         instance_config["mem"] = args.mem
+                    if any(
+                        (
+                            args.autotune_min_cpus,
+                            args.autotune_max_cpus,
+                            args.autotune_min_mem,
+                            args.autotune_max_mem,
+                            args.autotune_min_disk,
+                            args.autotune_max_disk,
+                        )
+                    ):
+                        instance_config["autotune"] = {}
```

Comment:
```suggestion
                        instance_config["autotune_limits"] = {}
```
is the key we want instead

---

might also be nice to do something like:
```python
limit_config = instance_config["autotune_limits"]
...
limit_config["cpus"]["min"] = args.autotune_min_cpus
...
```
so that we don't need to repeat `instance_config["autotune_limits"]` so many times :)

Reply:
i think maybe we could trim this code down by doing something like:
```
limit_config["cpus"] = {
    "min": args.autotune_min_cpus,
    "max": args.autotune_max_cpus
}
limit_config["mem"] = {
    "min": args.autotune_min_mem,
    "max": args.autotune_max_mem
}
limit_config["disk"] = {
    "min": args.autotune_min_disk,
    "max": args.autotune_max_disk
}```
and then adding a processing step that removes any min/max keys with `None` values - and any resource (e.g., cpus, mem, disk) keys with no subkeys

(that said, maybe the code to do that processing step wouldn't look too different and doing this would be a waste of time :p)